core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -164,6 +174,9 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
           driverConf.set(key, value)
         }
       }
+      // Periodically update the credentials for this user to ensure HDFS tokens get updated.
+      val tokenUpdater = new ExecutorDelegationTokenUpdater(driverConf, SparkHadoopUtil.get.conf)
+      tokenUpdater.updateCredentialsIfRequired()
```

Comment:
This will cause the code to go and check for the new credentials right away, right? Instead, how about having a `start()` method in ExecutorDelegationTokenUpdater that just schedules the task? (Then `updateCredentialsIfRequired` could be private.)